stan/math/prim/scal/prob/beta_rng.hpp

```diff
@@ -1,80 +1,84 @@
-      using std::exp;
       static const char* function = "beta_rng";
+
       check_positive_finite(function, "First shape parameter", alpha);
       check_positive_finite(function, "Second shape parameter", beta);
+      check_consistent_sizes(function, "First shape parameter", alpha,
+                             "Second shape Parameter", beta);
 
-      // If alpha and beta are large, trust the usual ratio of gammas
-      // method for generating beta random variables. If any parameter
-      // is small, work in log space and use Marsaglia and Tsang's trick
-      if (alpha > 1.0 && beta > 1.0) {
-        variate_generator<RNG&, gamma_distribution<> >
-          rng_gamma_alpha(rng, gamma_distribution<>(alpha, 1.0));
-        variate_generator<RNG&, gamma_distribution<> >
-          rng_gamma_beta(rng, gamma_distribution<>(beta, 1.0));
-        double a = rng_gamma_alpha();
-        double b = rng_gamma_beta();
-        return a / (a + b);
-      } else {
-        variate_generator<RNG&, uniform_real_distribution<> >
-          uniform_rng(rng, uniform_real_distribution<>(0.0, 1.0));
-        variate_generator<RNG&, gamma_distribution<> >
-          rng_gamma_alpha(rng, gamma_distribution<>(alpha + 1, 1.0));
-        variate_generator<RNG&, gamma_distribution<> >
-          rng_gamma_beta(rng, gamma_distribution<>(beta + 1, 1.0));
-        double log_a = log(uniform_rng()) / alpha + log(rng_gamma_alpha());
-        double log_b = log(uniform_rng()) / beta + log(rng_gamma_beta());
-        double log_sum = log_sum_exp(log_a, log_b);
-        return exp(log_a - log_sum);
+      scalar_seq_view<T_shape1> alpha_vec(alpha);
+      scalar_seq_view<T_shape2> beta_vec(beta);
+      size_t N = max_size(alpha, beta);
+      VectorBuilder<true, double, T_shape1, T_shape2> output(N);
+
+      variate_generator<RNG&, uniform_real_distribution<> >
+        uniform_rng(rng, uniform_real_distribution<>(0.0, 1.0));
+      for (size_t n = 0; n < N; ++n) {
+        // If alpha and beta are large, trust the usual ratio of gammas
+        // method for generating beta random variables. If any parameter
+        // is small, work in log space and use Marsaglia and Tsang's trick
+        if (alpha_vec[n] > 1.0 && beta_vec[n] > 1.0) {
+          variate_generator<RNG&, gamma_distribution<> >
+            rng_gamma_alpha(rng, gamma_distribution<>(alpha_vec[n], 1.0));
+          variate_generator<RNG&, gamma_distribution<> >
+            rng_gamma_beta(rng, gamma_distribution<>(beta_vec[n], 1.0));
+          double a = rng_gamma_alpha();
+          double b = rng_gamma_beta();
+          output[n] = a / (a + b);
+        } else {
+          variate_generator<RNG&, gamma_distribution<> >
+            rng_gamma_alpha(rng, gamma_distribution<>(alpha_vec[n] + 1, 1.0));
+          variate_generator<RNG&, gamma_distribution<> >
+            rng_gamma_beta(rng, gamma_distribution<>(beta_vec[n] + 1, 1.0));
+          double log_a = std::log(uniform_rng()) / alpha_vec[n]
```

Comment:
Optional.  I find it easier to read when the `+` is lined up under the `uniform_rng()` here so it's easy to scan this all within the log.